android-kit-base/src/main/java/com/mparticle/kits/KitManagerImpl.java

```diff
@@ -1523,13 +1523,23 @@ private void confirmEmail(
         boolean hasHashedEmail = hashedEmail != null && !hashedEmail.isEmpty();
 
         if ((hasEmail || hasHashedEmail) && user != null) {
+            MParticle.IdentityType selectedIdentityType = null;
+            try {
+                String identityTypeStr = (kitConfiguration != null)
+                        ? kitConfiguration.getHashedEmailUserIdentityType()
+                        : null;
+                if (identityTypeStr != null && !identityTypeStr.equalsIgnoreCase("Unknown")) {
+                    selectedIdentityType = MParticle.IdentityType.valueOf(identityTypeStr);
+                }
+            } catch (IllegalArgumentException e) {
+                selectedIdentityType = MParticle.IdentityType.Other; // fallback if invalid value
+            }
             String existingEmail = user.getUserIdentities().get(MParticle.IdentityType.Email);
-            String existingHashedEmail = user.getUserIdentities().get(MParticle.IdentityType.Other);
-
+            String existingHashedEmail = selectedIdentityType != null ? user.getUserIdentities().get(selectedIdentityType) : null;
             boolean emailMismatch = hasEmail && !email.equalsIgnoreCase(existingEmail);
             boolean hashedEmailMismatch = hasHashedEmail && !hashedEmail.equalsIgnoreCase(existingHashedEmail);
 
-            if (emailMismatch || hashedEmailMismatch) {
+            if (emailMismatch || (hashedEmailMismatch && selectedIdentityType != null)) {
```

Comment:
not sure if you will need `selectedIdentityType != null` anymore since it will be considered in order to get the `hashedEmailMismatch`, right?